examples/02_decoding/plot_miyawaki_encoding.py

```diff
@@ -0,0 +1,233 @@
+# Loading the data
+# ----------------
+# First some imports:
+
+import numpy as np
+import pylab as plt
+from nilearn.plotting import plot_stat_map
+from nilearn import masking
+
+
+##############################################################################
+# Now we can load the data set:
+from nilearn.datasets import fetch_miyawaki2008
+
+dataset = fetch_miyawaki2008()
+
+##############################################################################
+# We only use the training data of this study,
+# where random binary images were shown.
+
+X_random_filenames = dataset.func[12:]
+y_random_filenames = dataset.label[12:]
+
+# shape of the binary image in pixels
+y_shape = (10, 10)
+
+
+##############################################################################
+# We can use :func:`nilearn.input_data.MultiNiftiMaser` to load the fMRI data,
+# clean and mask it.
+
+from nilearn.input_data import MultiNiftiMasker
+
+masker = MultiNiftiMasker(mask_img=dataset.mask, detrend=True,
+                          standardize=True)
+masker.fit()
+X_train = masker.transform(X_random_filenames)
+
+# We load the visual stimuli from csv files
+y_train = []
+for y in y_random_filenames:
+    y_train.append(np.reshape(np.loadtxt(y, dtype=np.int, delimiter=','),
+                              (-1,) + y_shape, order='F'))
+
+
+##############################################################################
+# We now stack the X and y data and remove an offset in the beginning/end.
+
+X_train = np.vstack([x[2:] for x in X_train])
+y_train = np.vstack([y[:-2] for y in y_train]).astype(float)
+
+
+##############################################################################
+# X_train is a matrix of *N_samples* x *N_voxels*
+
+print(X_train.shape)
+
+##############################################################################
+# We flatten the last two dimensions of y_train
+# so it is a matrix of *N_samples* x *N_pixels*.
+
+# Flatten the stimuli
+y_train = np.reshape(y_train, (-1, y_shape[0] * y_shape[1]))
+
+print(y_train.shape)
+
+##############################################################################
+# Building the encoding models
+# ----------------------------
+# We can now proceed to do a simple encoding using Ridge regression.
+
+from sklearn.linear_model import Ridge
+from sklearn.cross_validation import KFold
+
+##############################################################################
+# Using 10-fold cross-validation, we partition the data into 10 'Folds',
+# we hold out each fold of the data for testing, then fit a ridge regression
+# to the remaining 9/10 of the data, using y_train as predictors
+# and X_train as targets, and create predictions for the held-out 10th.
+
+estimator = Ridge(alpha=100.)
+cv = KFold(len(y_train), 10)
+
+predictions = [Ridge(alpha=100.).fit(
+    y_train.reshape(-1, 100)[train], X_train[train]).predict(
+        y_train.reshape(-1, 100)[test])
+    for train, test in cv]
```

Comment:
I think cross_val_score needs a scorer that returns a single float, but we need an array of scores (one for each voxel). We could do cross_val_score independently for each voxel and then concatenate, but that makes the example much slower (7 minutes vs. 20 seconds for me)